compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -10563,69 +10558,53 @@ OMR::Z::TreeEvaluator::arraycmpHelper(TR::Node *node,
       // Threshold for when to generate a preemptive CLC to avoid pulling in 2 cache lines if the arrays differ towards the beginning
       uint32_t earlyCLCThreshold = 32;
 
-      isEndLabelNeeded = loopIters || needResultReg || length > (earlyCLCEnabled ? earlyCLCThreshold : TR_MAX_CLC_SIZE);
+      uint8_t numDeps = 0;
+
+      if (baseSource1Ref->getBaseRegister())
+            numDeps++;
 
-      if (length <= (earlyCLCEnabled ? earlyCLCThreshold : TR_MAX_CLC_SIZE) && (isFoldedIf || !needResultReg))
+      if (baseSource2Ref->getBaseRegister() && (baseSource2Ref->getBaseRegister() != baseSource1Ref->getBaseRegister()))
+            numDeps++;
+
+      if (isFoldedIf)
          {
-         // these are the only cases where no regDeps are needed as code will be just CLC/BRC with no internal control flow
-         regDeps = deps;
-         isCondLabelNeeded = false;
+         if (deps == NULL)
+            regDeps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, numDeps, cg);
+         else
+            regDeps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(deps, 0, numDeps, cg);
          }
       else
          {
-         uint8_t numDeps = 0;
-         // typically only 1 dep is needed but up to 4 could be needed so pre-compute to save space in most cases as regdep memory must be pre-allocated
-         if (length > (earlyCLCEnabled ? earlyCLCThreshold : TR_MAX_CLC_SIZE)) // NOTE: similar logic below when actually adding to regDeps
-            {
-            if (baseSource1Ref->getIndexRegister())
```

Comment:
no index register is allowed for ss instructions therefore this is not necessary.